remote/mattermost/remote.go

```diff
@@ -0,0 +1,191 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package mattermost
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+
+	"github.com/target/flottbot/models"
+	"github.com/target/flottbot/remote"
+)
+
+// Client struct.
+type Client struct {
+	Server   string
+	Token    string
+	BotID    string
+	Insecure bool
+}
+
+type mmLogger struct {
+	zerolog.Logger
+}
+
+func (l *mmLogger) Output(_ int, s string) error {
+	l.Logger.Info().Msg(s)
+	return nil
+}
+
+// validate that Client adheres to remote interface.
+var _ remote.Remote = (*Client)(nil)
+
+// instantiate a new mattermost client.
+func (c *Client) new() *model.Client4 {
+	log.Info().Msgf("%#v", c)
+
+	var url string
+	if c.Insecure {
+		url = "http://" + c.Server
+	} else {
+		url = "https://" + c.Server
+	}
```

Comment:
```suggestion
	url := "https://" + c.Server
	if c.Insecure {
		url = "http://" + c.Server
	}
```

i prefer the pattern you used for `wss`/`ws`